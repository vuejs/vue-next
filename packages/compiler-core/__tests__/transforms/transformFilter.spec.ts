import { CompilerOptions, generate, parse, transform } from '@vue/compiler-core'
import { transformFilter } from '../../src/transforms/transformFilter'
import { transformIf } from '../../src/transforms/vIf'
import { transformFor } from '../../src/transforms/vFor'
import { transformElement } from '../../src/transforms/transformElement'

function parseWithFilter(template: string, options: CompilerOptions = {}) {
  const ast = parse(template)

  transform(ast, {
    nodeTransforms: [
      transformIf,
      transformFor,
      transformFilter,
      transformElement
    ],
    ...options
  })

  return ast
}

describe('compiler: transform filter', () => {
  test('basic usage', () => {
    const ast = parseWithFilter('{{ msg | upper }}')

    expect(generate(ast).code).toMatchSnapshot()
  })

  test('chained usage', () => {
    const ast = parseWithFilter('{{ msg | upper | reverse}}')

    expect(generate(ast).code).toMatchSnapshot()
  })

  test('basic usage with element', () => {
    const ast = parseWithFilter('<div>{{ msg | upper }}</div>')

    expect(generate(ast).code).toMatchSnapshot()
  })

  test('basic usage with element v-bind', () => {
    const ast = parseWithFilter(`
				<div
          v-bind:id="id | upper | reverse"
          :class="cls | reverse"
          :ref="ref | lower">
        </div>`)

    expect(generate(ast).code).toMatchSnapshot()
  })
})
