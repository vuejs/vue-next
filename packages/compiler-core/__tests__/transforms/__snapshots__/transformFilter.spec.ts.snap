// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler: transform filter basic usage 1`] = `
"const _Vue = Vue

return function render() {
  with (this) {
    const { resolveFilter: _resolveFilter, toString: _toString } = _Vue
    
    const _filter_upper = _resolveFilter(\\"upper\\")
    return _toString(_filter_upper(msg))
  }
}"
`;

exports[`compiler: transform filter basic usage with element 1`] = `
"const _Vue = Vue

return function render() {
  with (this) {
    const { resolveFilter: _resolveFilter, toString: _toString, createVNode: _createVNode, createBlock: _createBlock, openBlock: _openBlock } = _Vue
    
    const _filter_upper = _resolveFilter(\\"upper\\")
    return (_openBlock(), _createBlock(\\"div\\", null, _toString(_filter_upper(msg)), 1 /* TEXT */))
  }
}"
`;

exports[`compiler: transform filter basic usage with element v-bind 1`] = `
"const _Vue = Vue

return function render() {
  with (this) {
    const { resolveFilter: _resolveFilter, createVNode: _createVNode, applyDirectives: _applyDirectives, resolveDirective: _resolveDirective, createBlock: _createBlock, openBlock: _openBlock } = _Vue
    
    const _directive_bind = _resolveDirective(\\"bind\\")
    const _filter_upper = _resolveFilter(\\"upper\\")
    const _filter_reverse = _resolveFilter(\\"reverse\\")
    const _filter_lower = _resolveFilter(\\"lower\\")
    
    return (_openBlock(), _applyDirectives(_createBlock(\\"div\\", null, null, 32 /* NEED_PATCH */), [
      [_directive_bind, _filter_reverse(_filter_upper(id)), \\"id\\"],
      [_directive_bind, _filter_reverse(cls), \\"class\\"],
      [_directive_bind, _filter_lower(ref), \\"ref\\"]
    ]))
  }
}"
`;

exports[`compiler: transform filter chained usage 1`] = `
"const _Vue = Vue

return function render() {
  with (this) {
    const { resolveFilter: _resolveFilter, toString: _toString } = _Vue
    
    const _filter_upper = _resolveFilter(\\"upper\\")
    const _filter_reverse = _resolveFilter(\\"reverse\\")
    return _toString(_filter_reverse(_filter_upper(msg)))
  }
}"
`;
